kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KopBrokerLookupManager.java

```diff
@@ -98,7 +100,11 @@ public KopBrokerLookupManager(KafkaServiceConfiguration conf, PulsarService puls
         if (log.isDebugEnabled()) {
             log.debug("Handle Lookup for topic {}", topicName);
         }
-        return LOOKUP_CACHE.computeIfAbsent(topicName, this::lookupBroker);
+
+        if (LOOKUP_CACHE.containsKey(topicName)) {
+            return LOOKUP_CACHE.get(topicName);
+        }
+        return lookupBroker(topicName);
```

Comment:
The composite operation of `containsKey` and `get` is not thread safe. We can use a single call of `get` here.

```suggestion
        final CompletableFuture<InetSocketAddress> future = LOOKUP_CACHE.get(topicName);
        return (future != null) ? future : lookupBroker(topicName);
```